### PR TITLE
Fix lifetime management of certificates with keys on macOS

### PR DESCRIPTION
**Pull request by vcsjones**
This is an attempt to address the certificate and key lifetime management issue reported in #94959.

The gist of the change is that now the certificate is up-refed and key along side with a public/private key pair. This seems to be the most reliable thing I was able to identify as a fix.

### CODE REVIEW

File: src/libraries/Common/src/System/Security/Cryptography/SecKeyPair.cs [hunk 1]
```diff
@@ -22,6 +24,9 @@ public void Dispose()
             PrivateKey = null;
             PublicKey?.Dispose();
             PublicKey = null!;
+            OwningCertificate?.Dispose();
```
Comment (filipnavara) [inline src/libraries/Common/src/System/Security/Cryptography/SecKeyPair.cs:27]: Doesn’t Dispose of SafeHandle have different characteristics from DangerousRelease? I find odd to pair it this way.

**Review by filipnavara — COMMENTED** [review]
The general idea looks sound to me.

File: src/libraries/Common/src/System/Security/Cryptography/SecKeyPair.cs (hunk 1 above)
Comment (vcsjones) [reply src/libraries/Common/src/System/Security/Cryptography/SecKeyPair.cs:27]: Ah, yeah, you're right. I started off duplicating but changed to adding a reference.

**Review by filipnavara — APPROVED** [review]

**Comment by vcsjones** [comment]
I think this is "close" but I want to re-do this in terms of safe handles. There are too many holes where this can potentially result in a native resource leak, I think.

I got what I needed from CI - I will re-open when ready.

**Pull request closed without merging**